src/core/theme/index.ts

```diff
@@ -1,73 +1,112 @@
 import './fontFaces.css';
+import { colors as colorTokens, shadows, gradients } from './colors';
 import {
-  colors as colorTokens,
-  shadows,
-  gradients,
-  outlines,
   ColorDesignTokens,
-} from './colors';
-import {
+  RawDesignTokens,
+  SpacingDesignTokens,
   suomifiDesignTokens,
   TypographyDesignTokens,
 } from 'suomifi-design-tokens';
-import {
-  spacing as spacingTokens,
-  SpacingProp,
-  SpacingDesignTokens,
-} from './spacing';
+import { spacing as spacingTokens, SpacingProp } from './spacing';
+import { boxShadowFocus, absoluteFocus, noMouseFocus } from './focus';
 import { zindexes } from './zindexes';
 import { transitions } from './transitions';
 import { radius } from './radius';
-export { SpacingProp };
 
-export type SuomifiTokens = typeof importedTokens;
-export type DefaultSuomifiTokens = typeof defaultThemeTokens;
-export type ColorProp = keyof typeof importedTokens.colors;
-export type SuomifiTheme = ReturnType<typeof suomifiTheme>;
+export {
+  TypographyDesignTokens,
+  ColorDesignTokens,
+  SpacingDesignTokens,
+  RawDesignTokens,
+  ValueUnit,
+  RawColorDesignTokens,
+  ColorToken,
+  RawSpacingDesignTokens,
+  RawTypographyDesignTokens,
+  TypographyToken,
+} from 'suomifi-design-tokens';
+
+export type GradientDesignTokens = typeof gradients;
+export type FocusDesignTokens = typeof derivedTokens.focus;
+export type RadiusDesignTokens = typeof radius;
+export type ShadowDesignTokens = typeof shadows;
+export type TransitionDesignTokens = typeof transitions;
 
+export type ColorProp = keyof typeof desingTokens.colors;
 export type TypographyProp = keyof typeof suomifiDesignTokens.typography;
+export { SpacingProp };
 
-export { TypographyDesignTokens } from 'suomifi-design-tokens';
+export type SuomifiDesignTokens = typeof desingTokens;
 
-export interface PartialTokens {
-  colors?: Partial<ColorDesignTokens>;
-  spacing?: Partial<SpacingDesignTokens>;
-  typography?: Partial<TypographyDesignTokens>;
+/**
+ * SuomifiTheme
+ * General styles and design tokens for suomifi-ui-components
+ */
+export interface SuomifiTheme {
+  /** Color design tokens as [key:string]: CSS string format with color name as key. */
+  colors: ColorDesignTokens;
+  /** Gradient design tokens as [key:string]: CSS string format with key derived from gradient name colors */
+  gradients: GradientDesignTokens;
```

Comment:
It is questionable if all of these should be called design tokens. Some are based on colors etc. but cannot / should not be directly derived by overridden colours due to various reasons. In a way gradients, shadows and transitions are similar to colors, spacing and typography => not derived, but smallest granular building blocks in the ds.